src/test/java/com/qunite/api/web/AuthenticationIntegrationTest.java

```diff
@@ -159,4 +174,99 @@ void signInShouldCheckPassword() throws Exception {
+
+    mockMvc.perform(delete("/users/self").header("authorization", token))
+        .andExpect(status().isNoContent());
+
+    mockMvc.perform(get("/users/self").header("authorization", token))
+        .andExpect(status().isForbidden());
+  }
+
+  @Test
+  @DisplayName("Token shouldn't work when username has been changed")
+  @Sql("/users-create.sql")
+  void updatedUsername() throws Exception {
+    var token = getAccessToken(1L);
+
+    var userUpdateDto = new UserUpdateDto();
+    userUpdateDto.setUsername("NEWUSERNAME");
+    var json = objectMapper.writeValueAsString(userUpdateDto);
+
+    mockMvc.perform(patch("/users/self").contentType(MediaType.APPLICATION_JSON)
+            .content(json).header(HttpHeaders.AUTHORIZATION, token))
+        .andExpect(status().isOk());
+
+    mockMvc.perform(get("/users/self").header(HttpHeaders.AUTHORIZATION, token))
+        .andExpect(status().isForbidden());
+  }
+
+  @Test
+  @DisplayName("Token should work when username hasn't been changed")
+  @Sql("/users-create.sql")
+  void notUpdatedUsername() throws Exception {
+    var token = getAccessToken(1L);
+
+    var userUpdateDto = new UserUpdateDto();
+    userUpdateDto.setEmail("NEWEMAIL@email.com");
+    var json = objectMapper.writeValueAsString(userUpdateDto);
+
+    mockMvc.perform(patch("/users/self").contentType(MediaType.APPLICATION_JSON)
+            .content(json).header(HttpHeaders.AUTHORIZATION, token))
+        .andExpect(status().isOk());
+
+    mockMvc.perform(get("/users/self").header(HttpHeaders.AUTHORIZATION, token))
+        .andExpect(status().isOk());
+  }
+
+  @Test
+  @DisplayName("Tokens should be invalidated when refresh token is used twice")
+  @Sql("/users-create.sql")
+  void refreshUse() throws Exception {
+    final var firstTokenPair = getTokenPair(1L);
+    final var secondTokenPair = getTokenPair(1L);
+    var body = new RefreshRequest();
+    body.setRefreshToken(firstTokenPair.getRefreshToken());
+    var json = objectMapper.writeValueAsString(body);
+
+    mockMvc.perform(
+            post("/{url}/sign-in/refresh", url).contentType(MediaType.APPLICATION_JSON)
+                .content(json))
+        .andExpect(status().isOk());
+    mockMvc.perform(
+            post("/{url}/sign-in/refresh", url).contentType(MediaType.APPLICATION_JSON)
+                .content(json))
+        .andExpect(status().isForbidden());
+
+    mockMvc.perform(
+            get("/users/self")
+                .header(HttpHeaders.AUTHORIZATION, secondTokenPair.getAccessToken()))
+        .andExpect(status().isForbidden());
+  }
+
+  @Test
+  @DisplayName("Tokens should be unique")
+  @Sql("/users-create.sql")
+  void uniqueTokenCheck() {
+    final var firstTokenPair = getTokenPair(1L);
+    final var secondTokenPair = getTokenPair(1L);
+
+    assertThat(firstTokenPair.getRefreshToken()).isNotEqualTo(secondTokenPair.getRefreshToken());
+    assertThat(firstTokenPair.getAccessToken()).isNotEqualTo(secondTokenPair.getAccessToken());
+  }
```

Comment:
Are you testing com.auth0.jwt method?

Reply:
If you testing only com.auth0.jwt.create() method, you can remove this test

Reply:
![image](https://github.com/q-unite/qunite-api/assets/103951082/2be11c60-8bdf-4f5f-a5ce-ef620d6429af)

I test that com.auth0.jwt.create() is used properly. Using it involves not just calling it, but adding additional fields as well, besides that, uniquity of token is not primary & obvious critery, so it should be tested separately.

Reply:
🤮🤮🤮🤮🤮🤮🤮🤮🤮